src/types.ts

```diff
@@ -821,6 +821,10 @@ export type MuteUserOptions<UserType = UnknownType> = {
 };
 
 export type PaginationOptions = {
+  created_at_after?: string | SeamlessImmutable.ImmutableDate | null;
+  created_at_after_or_equal?: string | SeamlessImmutable.ImmutableDate | null;
+  created_at_before?: string | SeamlessImmutable.ImmutableDate | null;
+  created_at_before_or_equal?: string | SeamlessImmutable.ImmutableDate | null;
```

Comment:
Tests were already added by @thesyncim here - https://github.com/GetStream/stream-chat-js/blob/master/test/integration/test.js#L3348

Reply:
Allowing `SeamlessImmutable.ImmutableDate` for following case:

```js
channel.query({
    messages: {
        created_at_before: channel.lastRead()
    }
})
```

Reply:
if we support `SeamlessImmutable.ImmutableDate` mayeb put `Date` there too?